packages/syft/src/syft/store/mongo_document_store.py

```diff
@@ -211,23 +232,43 @@ def _set(
     ) -> Result[SyftObject, str]:
         write_permission = ActionObjectWRITE(uid=obj.id, credentials=credentials)
         can_write = self.has_permission(write_permission)
+
+        store_query_key = self.settings.store_key.with_obj(obj)
+        collection_status = self.collection
+        if collection_status.is_err():
+            return collection_status
+        collection = collection_status.ok()
+
+        store_key_exists = (
+            collection.find_one({"_id": store_query_key.value}) is not None
+        )
+
+        if not store_key_exists:
+            # attempt to claim ownership for writing
+            ownership_result = self.take_ownership(uid=obj.id, credentials=credentials)
+            can_write: bool = ownership_result.is_ok()
+
         if can_write:
             storage_obj = obj.to(self.storage_type)
 
-            collection_status = self.collection
-            if collection_status.is_err():
-                return collection_status
-            collection = collection_status.ok()
-
             if ignore_duplicates:
                 collection = collection.with_options(write_concern=WriteConcern(w=0))
             try:
                 collection.insert_one(storage_obj)
             except DuplicateKeyError as e:
                 return Err(f"Duplicate Key Error for {obj}: {e}")
+
```

Comment:
not sure if the logic here is completely the same as the logic in the kv_docstore, (althought that logic also seems a bit flawed. In the kv_doc store, if the item already exists and we do ignore duplicates, we would **not write**, but return the original object instead. I am not saying that that logic makes sense, but it looks like there is a discrepancy between that and this logic

Reply:
when not setting `WriteConcern(w=0)`, an error will be raised if the item already exists.
When we set  WriteConcern(w=0)) , adding existed items will not raise error, and the database will not be changed (so it kept the existed item)

Reply:
I resolved this in the latest commits, please have a look @khoaguin @shubham3121